python/src/aiconfig/callback.py

```diff
@@ -0,0 +1,139 @@
+# Standard Libraries
+import asyncio
+import json
+import time
+
+# Third Party Libraries
+from result import Err, Ok
+from pydantic import BaseModel, ConfigDict
+from typing import (
+    Coroutine,
+    Final,
+    List,
+    Sequence,
+    TextIO,
+    TypeAlias,
+    Union,
+    Any,
+    Awaitable,
+    Callable,
+    Optional,
+)
+
+# Constants
+DEFAULT_TIMEOUT = 5  # Default timeout for callback execution in seconds
+
+
+class CallbackEvent:
+    """
+    Represents an event with data to be passed to a callback.
+    """
+
+    def __init__(self, name: str, data: Any, ts_ns: int = time.time_ns()):
+        self.name = name
+        # Anything available at the time the event happens.
+        # It is passed to the callback.
+        self.data = data
+        self.ts_ns = ts_ns
+
+
+# Type Aliases
+Callback = Callable[[CallbackEvent], Awaitable]
+Result: TypeAlias = Union[Ok[Any], Err[Any]]
+
+
+async def execute_coroutine_with_timeout(
+    coroutine: Coroutine[Any, Any, Any], timeout: int
+) -> Result:
```

Comment:
this also needs to be parametrized. Pyright catches this for me: do you have strict config?

Reply:
could you share your Pyright setup? I've been rolling with the stand pylint checker and haven't had it complain

Reply:
https://github.com/lastmile-ai/semantic-retrieval/blob/main/pyrightconfig.json

Strict with a couple of specific checks I disabled, I think because they were impossible to reconcile with numpy.

It's probably better to start as strict as we want to get, early, because going the opposite direction will be harder.

Reply:
@LastMileNewGrad , this is not blocking